### DISCUSSION TITLE
Required context role question

### DISCUSSION THREAD

**Discussion by angelicochris** [w3c/aria#1864] [Q&A] [answered]
We currently use Foundation Site for our website and their implementation of tabs seems to be flagging a rule violation in site improve related to the required context role. I can't seem to find anywhere that gives a straight answer if such an implementation is valid or not. Here is an example from their tabs pages.

```html
<ul class="tabs" data-tabs="ob5o6y-tabs" id="example-tabs" role="tablist">
  <li class="tabs-title is-active" role="presentation"><a href="#panel1" aria-selected="true" role="tab" aria-controls="panel1" id="panel1-label" tabindex="0">Tab 1</a></li>
  <li class="tabs-title" role="presentation"><a href="#panel2" role="tab" aria-controls="panel2" aria-selected="false" id="panel2-label" tabindex="-1">Tab 2</a></li>
  <li class="tabs-title" role="presentation"><a href="#panel3" role="tab" aria-controls="panel3" aria-selected="false" id="panel3-label" tabindex="-1">Tab 3</a></li>
  <li class="tabs-title" role="presentation"><a href="#panel4" role="tab" aria-controls="panel4" aria-selected="false" id="panel4-label" tabindex="-1">Tab 4</a></li>
  <li class="tabs-title" role="presentation"><a href="#panel5" role="tab" aria-controls="panel5" aria-selected="false" id="panel5-label" tabindex="-1">Tab 5</a></li>
  <li class="tabs-title" role="presentation"><a href="#panel6" role="tab" aria-controls="panel6" aria-selected="false" id="panel6-label" tabindex="-1">Tab 6</a></li>
</ul>
```

**Comment by scottaohara** [comment, marked as answer]
maybe a question for site improve why they are marking this as invalid.  there's no issue here, based on the markup provided.  role=presentation double negates the li elements' list item semantics (it's also already negated due to the fact that the ul has a role=tablist which already means the li elements lose their list item semantics.  so, as far as the a11y tree is concerned, there's a direct parent/child relationship between the tablist and its tab descendants.

**Reply by angelicochris** [reply]
Here is our versions that is getting flagged.
![image](https://user-images.githubusercontent.com/55987158/215167644-59d89975-3dd2-42fc-8d20-e5e2a3eca05d.png)

**Reply by scottaohara** [reply]
well there you go then, very different details than what you posted.  As the ARIA spec notes, there's a tabindex=0 on the list item that then contains the tab element.  Making a role=presentation focusable UNDOES the role=presentation. If the guidance as provided by the checker was not clear as to why there was an issue here, then it again seems this should be surfaced back to the checker.

**Reply by angelicochris** [reply]
Ok and by undoing what presentation was doing it brings back the require context role issue. Yea site improve only mentioned there was an issue with a role not being in a required context:
![image](https://user-images.githubusercontent.com/55987158/215182676-2fb95a29-4188-4917-92bc-aa2381a8caf1.png)
Guess it missed giving the why in this instance.
